grails-app/controllers/com/unifina/controller/api/KeyApiController.groovy

```diff
@@ -126,22 +127,53 @@ class KeyApiController {
 	}
 
 	@StreamrApi(authenticationLevel = AuthLevel.USER)
-	def update() {
-		Key k = Key.get(params.id)
+	def updateUserKey() {
+		Key k = Key.get(params.keyId)
 		if (k == null) {
-			throw new NotFoundException(k.toString(), params.id)
+			throw new NotFoundException(k.toString(), params.keyId)
 		}
 		Map json = new JsonSlurper().parseText((String) request.JSON)
 		if (json.name != null && json.name.trim() != "") {
-			useResource(params.resourceClass, params.resourceId) { res ->
+			useResource(SecUser, params.keyId) { res ->
 				k.name = json.name.trim()
 				k.save(flush: true, failOnError: true)
 			}
 		}
 		response.status = 200
-		render([
-			id: k.id,
-			name: k.name,
-		] as JSON)
+		render(k.toMap() as JSON)
+	}
+
+	@StreamrApi(authenticationLevel = AuthLevel.USER)
+	def updateStreamKey() {
+		Key k = Key.get(params.keyId)
+		if (k == null) {
+			throw new NotFoundException(k.toString(), params.keyId)
+		}
+		Map json = new JsonSlurper().parseText((String) request.JSON)
+		String permission = request.JSON?.permission
+		if (permission) {
```

Comment:
Two `if`s can be combined:
`if (permission && permission != "read" && permission != "write") {`